src/west/commands/common.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2019, Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+'''West common commands'''
+
+import argparse
+import textwrap
+
+from west.commands import WestCommand
+from west.util import WestReload
+
+class Help(WestCommand):
+    def __init__(self):
+        super().__init__(
+            'help',
+            'print west command help text',
+            _wrap('''
+            Print help text on any command.
+            '''))
+
+    def do_add_parser(self, parser_adder):
+        return _add_parser(
+            parser_adder, self,
+            _arg('command', metavar='cmd', nargs='?',
+                 help='''Command to get help on''')
+        )
+
+    def do_run(self, args, user_args):
+        a = [args.command, '-h']
+        wr = WestReload(argv=[x for x in a if x is not None])
+        raise wr
+
+def _arg(*args, **kwargs):
+    # Helper for creating a new argument parser for a single argument,
+    # later passed in parents= to add_parser()
+
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument(*args, **kwargs)
+    return parser
+
+
+# Arguments shared between more than one command
+
+def _add_parser(parser_adder, cmd, *extra_args, **kwargs):
```

Comment:
Is it worth moving _arg, _add_parser, and _wrap to helpers importable from `west.command` instead of keeping separate copies in project.py and this new file?

Reply:
Yes, I think this makes sense. I will take a look.

Reply:
moved